datacube/utils/aws/__init__.py

```diff
@@ -438,3 +438,15 @@ def get_aws_settings(profile: Optional[str] = None,
                  aws_secret_access_key=cc.secret_key,
                  aws_session_token=cc.token,
                  requester_pays=requester_pays), creds)
+
+
+def obtain_new_IAM_auth_token(url: str, region_name: str = "auto", profile_name: Optional[str] = None) -> str:
```

Comment:
- `url: URL`, not str
- lowercase function name for consistency with the rest of ODC (I don't think there's strict pep8 linter enabled, but it's convention elsewhere in the repo)

Reply:
Thanks that's really helpful.